cpu/cortexm_common/include/irq_arch.h

```diff
@@ -18,24 +18,33 @@
  * @}
  */
 
+
+#ifndef _IRQ_ARCH_H
+#define _IRQ_ARCH_H
+
 #include <stdint.h>
-#include "irq.h"
-#include "cpu.h"
+#include "cpu_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 /**
  * @brief Disable all maskable interrupts
  */
-unsigned int irq_disable(void)
+static inline __attribute__((always_inline)) unsigned int irq_disable(void)
 {
     uint32_t mask = __get_PRIMASK();
+
     __disable_irq();
     return mask;
 }
 
 /**
  * @brief Enable all maskable interrupts
  */
-__attribute__((used)) unsigned int irq_enable(void)
+static inline __attribute__((always_inline)) __attribute__((used)) unsigned int
+irq_enable(void)
 {
     __enable_irq();
     return __get_PRIMASK();
```

Comment:
Mind to sneak in an unrelated bugfix as separate commit? It should be:

```C
unsigned result = __get_PRIMASK();
__enable_irq();
return result;
```

(See [API doc](http://api.riot-os.org/group__core__irq.html#gac5652462dfaa82609d3f88f6b09519ae))

Reply:
done